### PR TITLE
Add map_index to Log model

### PR DESCRIPTION
**Pull request by uranusjr**
This is much easier than most since the fields in Log are only informational and do not actually establish relationships to TaskInstance.

I also chose to keep execution_date for now since it does not really matter to adding map_index on Log (unlike many other models). Perhaps we can have both execution_date and run_id? That can be a decision for later.

### CODE REVIEW

File: airflow/migrations/versions/0105_75d5ed6c2b43_add_map_index_to_log.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+
+def downgrade():
+    """Remove map_index from Log."""
+    conn = op.get_bind()
+    with op.batch_alter_table("log") as batch_op:
+        if conn.dialect.name == "mssql":
+            for name in _find_map_index_constraints(conn):
+                batch_op.drop_constraint(name)
+        batch_op.drop_column("map_index")
```
Comment (ephraimbuddy) [inline airflow/migrations/versions/0105_75d5ed6c2b43_add_map_index_to_log.py]:
```suggestion
        batch_op.drop_column("map_index", mssql_drop_default=True)
```

File: airflow/migrations/versions/0105_75d5ed6c2b43_add_map_index_to_log.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+def upgrade():
+    """Add map_index to Log."""
+    op.add_column("log", Column("map_index", Integer, server_default="-1"))
+    # Only set map_index if the log is related to a task instance.
+    op.execute(
+        """
+        update log set map_index = -1
+        where dag_id is not null and task_id is not null and execution_date is not null
+        """
```
Comment (ephraimbuddy) [inline airflow/migrations/versions/0105_75d5ed6c2b43_add_map_index_to_log.py]: I think that the `server_default` above will set the `map_index` regardless of whether dag_id etc is null

File: airflow/migrations/versions/0105_75d5ed6c2b43_add_map_index_to_log.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    result = conn.execute(
+        """
+        SELECT CONSTRAINT_NAME
+        FROM INFORMATION_SCHEMA.CONSTRAINT_COLUMN_USAGE
+        WHERE TABLE_NAME = 'log' AND COLUMN_NAME = 'map_index'
+        """,
+    )
+    if result:
+        return result.fetchall()
+    return []
```
Comment (ephraimbuddy) [inline airflow/migrations/versions/0105_75d5ed6c2b43_add_map_index_to_log.py]:
```suggestion
```
seems not used?

File: airflow/models/log.py [hunk 4]
```diff
@@ -30,8 +30,9 @@ class Log(Base):
 
     id = Column(Integer, primary_key=True)
     dttm = Column(UtcDateTime)
-    dag_id = Column(String(ID_LEN, **COLLATION_ARGS))
-    task_id = Column(String(ID_LEN, **COLLATION_ARGS))
+    dag_id = Column(StringID())
+    task_id = Column(StringID())
+    map_index = Column(Integer, server_default='-1')
```
Comment (ephraimbuddy) [inline airflow/models/log.py]: If we do not want the map_index value to be set when dag_id, task_id, etc are null, I think we shouldn't use `server_default`. We could use default: 
```python
map_index = Column(Integer, default=lambda:set_map_index())
```
Then workout the map_index in `set_map_index`?

File: airflow/migrations/versions/0105_75d5ed6c2b43_add_map_index_to_log.py (hunk 3 above)
Comment (uranusjr) [reply airflow/migrations/versions/0105_75d5ed6c2b43_add_map_index_to_log.py]: Right, it’s a relic from previous iterations. Thanks.

File: airflow/models/log.py (hunk 4 above)
Comment (uranusjr) [reply airflow/models/log.py]: Or we could just not have a default at all! The server_default was added on some models to make migration faster, but that’s less of a goal here.

**Review by ephraimbuddy — APPROVED** [review]

**Pull request merged**